addons/website/static/src/snippets/s_share/000.js

```diff
@@ -75,12 +74,22 @@ const ShareWidget = publicWidget.Widget.extend({
                     return `${a + title}%20${url + c}`;
                 }
                 return a + title + c;
-            })
-            .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
             });
+        const urlObject = new URL(aEl.href);
+        if (urlObject.searchParams.has("media")) {
+            const ogImageEl = document.querySelector("meta[property='og:image']");
+            // Some pages (/profile/user/ID) don't have an image to share.
+            if (ogImageEl) {
+                const media = encodeURIComponent(ogImageEl.content);
+                urlObject.searchParams.set("media", media);
```

Comment:
Technically, I'd say that we'd still need to replace the media parameter on the `<a>` href to stay stable... but let's not.